code/datums/components/barber.dm

```diff
@@ -0,0 +1,605 @@
+		user.visible_message("[user] assumes a less hostile grip on the [thing].",\
+													"You change your grip on the [thing], so as to use it more as a tool than a weapon.")
+	else
+		user.visible_message("[user] wields the [thing] a with [pick(cool_grip_adj)] [pick(cool_grip1)] [pick(cool_grip2a)][pick(cool_grip2b)] [pick("style", "grip")] that they probably [pick(cool_grip3)]!",\
+													"You wield the [thing] with [pick(cool_grip_adj)] [pick(cool_grip1)] [pick(cool_grip2a)][pick(cool_grip2b)] [pick("style", "grip")] that you [pick(cool_grip3)]! It makes it just about impossible to use as a tool!")
+
+/datum/component/toggle_tool_use/UnregisterFromParent()
+	UnregisterSignal(parent, COMSIG_ITEM_ATTACK_SELF)
+	UnregisterSignal(parent, COMSIG_ITEM_DROPPED)
+	. = ..()
+
+/datum/component/barber
+/datum/component/barber/Initialize()
+	if(!istype(parent, /obj/item))
+		return COMPONENT_INCOMPATIBLE
+
+/datum/component/barber/haircut
+/datum/component/barber/haircut/Initialize()
+	RegisterSignal(parent, list(COMSIG_ITEM_ATTACK_PRE), .proc/do_haircut)
+
+/datum/component/barber/shave
+/datum/component/barber/shave/Initialize()
+	RegisterSignal(parent, list(COMSIG_ITEM_ATTACK_PRE), .proc/do_shave)
+
+/datum/component/barber/proc/do_haircut(var/obj/item/thing, mob/living/carbon/human/M as mob, mob/living/carbon/human/user as mob)
+	if(!M || !user || (user.a_intent != INTENT_HELP && !thing.force_use_as_tool))
+		return 0 // Who's cutting whose hair, now?
+
+	var/mob/living/carbon/human/H = M
+	if(ishuman(M) && ((H.head && H.head.c_flags & COVERSEYES) || (H.wear_mask && H.wear_mask.c_flags & COVERSEYES) || (H.glasses && H.glasses.c_flags & COVERSEYES)))
+		// you can't stab someone in the eyes wearing a mask!
+		boutput(user, "<span class='notice'>You're going to need to remove that mask/helmet/glasses first.</span>")
+		. = BARBERY_RESOLVABLE
+
+	if(M.bioHolder.mobAppearance.customization_first == "None")
+		boutput(user, "<span class='alert'>There is nothing to cut!</span>")
+		. = BARBERY_RESOLVABLE
+
+	if(!mutant_barber_fluff(M, user, "haircut"))
+		return BARBERY_SUCCESSFUL
+
+	if(.)
+		if (thing.force_use_as_tool || (user.a_intent == INTENT_HELP && (istype(M.buckled, /obj/stool/chair/comfy/barber_chair) || istype(get_area(M), /area/station/crew_quarters/barber_shop))))
+			if (. != BARBERY_RESOLVABLE)
+				boutput(user, "<span class='notice'>You poke [M] with your [thing]. If you want to attack [M], you'll need to remove [him_or_her(M)] from the barber shop or set your intent to anything other than 'help', first.</span>")
+			return ATTACK_PRE_DONT_ATTACK
+		else
+			return 0
+
+	SPAWN_DBG(0)
+		var/new_style = input(user, "Please select style", "Style")  as null|anything in customization_styles + customization_styles_gimmick
+
+		if (new_style)
+			if(M.bioHolder.mobAppearance.customization_first == "Balding" && new_style != "None")
+				boutput(user, "<span class='alert'>Not enough hair!</span>")
+				return ATTACK_PRE_DONT_ATTACK
```

Comment:
Note that this is in a spawn so it won't actually return the value. I recommend replacing just with 
```suggestion
				return
```